pypeit/reduce.py

```diff
@@ -276,19 +276,20 @@ def extract(self, global_sky, sobjs_obj):
             self.sobjs = self.sobjs_obj.copy()
             # Purge out the negative objects if this was a near-IR reduction unless negative objects are requested
 
```

Comment:
I don't understand the control flow here and the addition of this skip_optimal parameter. Can you please elaborate. The the user is skipping optimal, and also skipping boxcar, then it is unclear to me what this method is doing. Basically it is just filling things with zeros. If that is the case, this should just be done outside of the extract, i.e. you are not extracting anything so why is the extract method being called in the control flow? In other words, skip_boxcar is not an appropriate name for the parameter. It should be "skip_extraction" and this if then statement should go somewhere in reduce.py before extract is called to prevent extraction and simply fill everything with zeros.